wandb/sdk/launch/runner/kubernetes_runner.py

```diff
@@ -715,30 +718,37 @@ def add_wandb_env(root: Union[dict, list], env_vars: Dict[str, str]) -> None:
     Recursively walks the spec and injects the environment variables into
     every container spec. Containers are identified by the "containers" key.
 
+    This function treats the WANDB_RUN_ID and WANDB_GROUP_ID environment variables
+    specially. If they are present in the spec, they will be overwritten. If a setting
+    for WANDB_RUN_ID is provided in env_vars, then that environment variable will only be
+    set in the first container modified by this function.
+
     Arguments:
         root: The spec to modify.
         env_vars: The environment variables to inject.
 
     Returns: None.
     """
-    if isinstance(root, dict):
-        for k, v in root.items():
-            if k == "containers":
-                if isinstance(v, list):
-                    for cont in v:
-                        env = cont.get("env", [])
-                        env.extend(
-                            [
-                                {"name": key, "value": value}
-                                for key, value in env_vars.items()
-                            ]
-                        )
-                        cont["env"] = env
-            elif isinstance(v, (dict, list)):
-                add_wandb_env(v, env_vars)
-    elif isinstance(root, list):
-        for item in root:
-            add_wandb_env(item, env_vars)
+
+    def yield_containers(root):
+        if isinstance(root, dict):
+            for k, v in root.items():
+                if k == "containers":
+                    if isinstance(v, list):
+                        for cont in v:
+                            yield cont
+                elif isinstance(v, (dict, list)):
+                    yield from yield_containers(v)
+        elif isinstance(root, list):
+            for item in root:
+                yield from yield_containers(item)
+
+    for cont in yield_containers(root):
+        env = cont.setdefault("env", [])
+        env.extend([{"name": key, "value": value} for key, value in env_vars.items()])
+        cont["env"] = env
+        if "WANDB_RUN_ID" in env_vars:
+            env_vars.pop("WANDB_RUN_ID")
```

Comment:
So this removes the WANDB_RUN_ID after the first container, is the first container guaranteed to be the order 0 container? Also, can we add a comment to make this more clear about why we're doing this?